src/config.c

```diff
@@ -217,26 +223,32 @@ static RedisModuleString *getString(const char *name, void *privdata)
          * previous value and assign a new config object we are about to return.
          * Related issue: https://github.com/redis/redis/issues/10749
          */
-        RedisModule_FreeString(NULL, c->tmp);
-        c->tmp = ret;
+        RedisModule_FreeString(NULL, c->str_conf_ref);
+        c->str_conf_ref = ret;
     }
 
     return ret;
 }
 
-static int setString(const char *name, RedisModuleString *val,
-                     void *privdata, RedisModuleString **err)
+static int setString(const char *name,
+                     RedisModuleString *val,
+                     void *privdata,
+                     RedisModuleString **err)
 {
+    RedisRaftCtx *rr = &redis_raft;
     RedisRaftConfig *c = privdata;
 
-    const char *errmsg = c->before_cb(c->arg, name);
-    if (errmsg) {
-        *err = RedisModule_CreateString(NULL, errmsg, strlen(errmsg));
-        return REDISMODULE_ERR;
+    if (rr->state != REDIS_RAFT_UNINITIALIZED) {
```

Comment:
actually I think `before_cb`\`after_cb` mechanism was good, and enable us expanding it when needed + have all the config rules in one place. But as you wish.

Reply:
I realized we also need the value of the config change depending on the config. So, callback won't be simple anymore. We'd need variadic arguments.